src/gui/map/map_find_feature.cpp

```diff
@@ -198,23 +198,23 @@ void MapFindFeature::findNext()
 	Object* next_match = nullptr;   // the next match after pivot_object
 	map->clearObjectSelection(false);
 	
-	auto search = [&first_match, &pivot_object, &next_match](Object* object) {
+	auto search = [&](Object* object) {
 		if (next_match)
 			return;
 		
 		bool after_pivot = (pivot_object == nullptr);
 		if (object == pivot_object)
 			pivot_object = nullptr;
 		
-		if (isSelectable(object))
+		if (isSelectable(object) && query(object))
 		{
 			if (after_pivot)
 				next_match = object;
 			else if (!first_match)
 				first_match = object;
 		}
 	};
-	map->getCurrentPart()->applyOnMatchingObjects(search, std::cref(query));
+	map->getCurrentPart()->applyOnAllObjects(search);
```

Comment:
Reverting part of my change here: The current object might no longer match the query, but we still want to see it when determining the next match.